### PR TITLE
Add optional ResourceGroupName to LTR cmdlet

### PR DESCRIPTION
**Pull request by pixia**
## Checklist

- [x] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#updating-the-change-log)
- [x] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [x] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (Microsoft internal partners only)
    - [x] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/help-generation.md)

### CODE REVIEW

File: src/Sql/Sql/Database Backup/Services/AzureSqlDatabaseBackupAdapter.cs [hunk 1]
```diff
@@ -390,24 +392,36 @@ private AzureSqlDatabaseLongTermRetentionBackupModel GetBackupModel(Management.S
                 Location = locationName,
                 ResourceId = backup.Id,
                 ServerCreateTime = backup.ServerCreateTime,
-                ServerName = backup.ServerName
+                ServerName = backup.ServerName,
+                ResourceGroupName = GetResourceGroupNameFromResourceId(backup.Id)
             };
         }
 
+        private string GetResourceGroupNameFromResourceId(string resourceId)
```
Comment (johnpaulkee) [inline src/Sql/Sql/Database Backup/Services/AzureSqlDatabaseBackupAdapter.cs]: There is a ResourceIdentifer helper class that you can use to help grab ResourceGroupName from resourceId.

File: src/Sql/Sql/Database Backup/Cmdlet/RemoveAzureRmSqlDatabaseLongTermRetentionBackup.cs [hunk 2]
```diff
@@ -195,5 +201,37 @@ public override void ExecuteCmdlet()
                 }
             }
         }
+
+        /// <summary>
+        /// Parse the longTermRetentionBackup resource Id
+        /// </summary>
+        /// <param name="resourceId"></param>
+        private void ParseLongTermRentionBackupResourceId(string resourceId)
```
Comment (johnpaulkee) [inline src/Sql/Sql/Database Backup/Cmdlet/RemoveAzureRmSqlDatabaseLongTermRetentionBackup.cs]: Consider using ResourceIdentifier class here.

**Comment by pixia** [comment]
> Approved, just some suggestions to use ResourceIdentifier class to parse resource id.

As discussed offline, ResourceIdentifier does not fit my needs. Will stick to the current implementation.

File: src/Sql/Sql.Test/ScenarioTests/DatabaseBackupTests.cs [hunk 3]
```diff
@@ -102,6 +115,14 @@ public void TestLongTermRetentionV2()
                 RunPowerShellTest("Test-LongTermRetentionV2");
             }
         }
+
+        [Fact(Skip = "This is not recordable test")]
```
Comment (markcowl) [inline src/Sql/Sql.Test/ScenarioTests/DatabaseBackupTests.cs]: Please do not add a skipped test.

File: src/Sql/Sql.Test/ScenarioTests/DatabaseBackupTests.ps1 [hunk 4]
```diff
@@ -255,6 +283,61 @@ function Test-LongTermRetentionV2
 	Remove-AzSqlDatabase -ResourceGroup $resourceGroup -ServerName $serverName -DatabaseName $restoredDatabase
 }
 
+function Test-LongTermRetentionV2ResourceGroupBased
```
Comment (markcowl) [inline src/Sql/Sql.Test/ScenarioTests/DatabaseBackupTests.ps1]: do not add a skipped test

File: src/Sql/Sql.Test/ScenarioTests/DatabaseBackupTests.ps1 [hunk 5]
```diff
@@ -255,6 +283,61 @@ function Test-LongTermRetentionV2
...
+{
+
+	# MANUAL INSTRUCTIONS
+	# Playback mode
+	#    Remove skip flag from test
+	# Record mode
+	#    Create a server and database and fill in the appropriate information below
+	#    Set the weekly retention on the database so that the first backup gets picked up, for example:
+	#    Set-AzSqlDatabaseLongTermRetentionPolicy -ResourceGroup $resourceGroup -ServerName $serverName -DatabaseName $databaseName -WeeklyRetention P1W
```
Comment (markcowl) [inline src/Sql/Sql.Test/ScenarioTests/DatabaseBackupTests.ps1]: Just create the necessary items as part of the test

File: src/Sql/Sql/Database Backup/Cmdlet/RemoveAzureRmSqlDatabaseLongTermRetentionBackup.cs [hunk 6]
```diff
@@ -107,6 +108,16 @@ public class RemoveAzureRmSqlDatabaseLongTermRetentionBackup : AzureSqlDatabaseL
         [ValidateNotNullOrEmpty]
         public string BackupName { get; set; }
 
+        /// <summary>
+        /// Gets or sets the name of the resource group to use.
+        /// </summary>
+        [Parameter(Mandatory = false,
+            ParameterSetName = RemoveBackupDefaultSet,
+            Position = 0,
```
Comment (markcowl) [inline src/Sql/Sql/Database Backup/Cmdlet/RemoveAzureRmSqlDatabaseLongTermRetentionBackup.cs]: optional parameters should not have positions and definitely should not be position  0

File: src/Sql/Sql/Database Backup/Cmdlet/GetAzureRmSqlDatabaseLongTermRetentionBackup.cs [hunk 7]
```diff
@@ -169,6 +169,24 @@ public class GetAzureRmSqlDatabaseLongTermRetentionBackup : AzureSqlDatabaseLong
...
+            ParameterSetName = LocationSet,
+            Position = 0,
+            HelpMessage = "The name of the resource group.")]
+        [Parameter(Mandatory = false,
+            ParameterSetName = ServerNameSet,
+            Position = 0,
+            HelpMessage = "The name of the resource group.")]
+        [Parameter(Mandatory = false,
+            ParameterSetName = BackupNameSet,
+            Position = 0,
```
Comment (markcowl) [inline src/Sql/Sql/Database Backup/Cmdlet/GetAzureRmSqlDatabaseLongTermRetentionBackup.cs]: Do not add a position to an optional parameter, and certainly not position 0

File: src/Sql/Sql.Test/ScenarioTests/DatabaseBackupTests.cs (hunk 3 above)
Comment (pixia) [reply src/Sql/Sql.Test/ScenarioTests/DatabaseBackupTests.cs]: I have the test pass and recorded. However the test set up (i.e. ensure a backup exist) need at least 18 hours. That is why I marked it as skipped.

Comment (markcowl) [reply src/Sql/Sql.Test/ScenarioTests/DatabaseBackupTests.cs]: so, if the test takes 18 hours, you should not check it in.  Please remove it.

**Comment by cormacpayne** [comment]
@pixia some of the Sql tests are failing as a part of this PR -- would you mind taking a look?

File: src/Sql/Sql/Database Backup/Cmdlet/GetAzureRmSqlDatabaseLongTermRetentionBackup.cs (hunk 7 above)
Comment (pixia) [reply src/Sql/Sql/Database Backup/Cmdlet/GetAzureRmSqlDatabaseLongTermRetentionBackup.cs]: fixed.

**Pull request merged**